front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx

```diff
@@ -328,7 +292,7 @@ const RollingStockEditorForm = ({
     },
   });
 
-  const addNewRollingstock = (data: FieldValues) => {
+  const addNewRollingstock = (data: RollingStockParametersValues) => {
     postRollingstock({
       locked: false,
       rollingStockUpsertPayload: queryArg(data),
```

Comment:
I tested and the catch doesn't work
```suggestion
  const addNewRollingstock = (data: RollingStockParametersValues) => {
    postRollingstock({
      locked: false,
      rollingStockUpsertPayload: queryArg(data),
    })
      .unwrap()
      .then(
        () => {
          dispatch(
            setSuccess({
              title: t('messages.success'),
              text: t('messages.rollingStockAdded'),
            })
          );
          setAddOrEditState(false);
        },
        (error) => {
          if (error.data?.message.includes('duplicate')) {
            dispatch(
              setFailure({
                name: t('messages.failure'),
                message: t('messages.rollingStockDuplicateName'),
              })
            );
          } else {
            dispatch(
              setFailure({
                name: t('messages.failure'),
                message: t('messages.rollingStockNotAdded'),
              })
            );
          }
        }
      )
```

Reply:
are you sure? I just tested and it worked for me

Reply:
Tested and yes it works !